File: spring-webflux-java/src/main/java/com/example/springwebfluxjava/router/ApiRouterConfig.java
package com.example.springwebfluxjava.router;

import com.example.springwebfluxjava.handler.ApiHandler;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.HttpStatus;
import org.springframework.web.reactive.function.client.WebClientResponseException;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.RouterFunctions;
import org.springframework.web.reactive.function.server.ServerResponse;
import org.springframework.web.server.WebFilter;

import static org.springframework.web.reactive.function.server.RequestPredicates.GET;
import static org.springframework.web.reactive.function.server.RequestPredicates.path;
import static org.springframework.web.reactive.function.server.RouterFunctions.route;

@Configuration
public class ApiRouterConfig {

    @Bean
    public RouterFunction<ServerResponse> apiRouter(ApiHandler apiHandler) {
        return RouterFunctions.nest(path("/api"),
                route(GET("/anime/{id}"), apiHandler::getAnimeAndMovie)
                        .andRoute(GET("/movie"), apiHandler::getMovies));
    }

    @Bean
    WebFilter handleExceptions() {
        return (exchange, next) -> next.filter(exchange)
                .onErrorResume(e -> {
                    var response = exchange.getResponse();
                    if (e instanceof WebClientResponseException ex) {
                        response.setStatusCode(ex.getStatusCode());
                    } else {
                        response.setStatusCode(HttpStatus.INTERNAL_SERVER_ERROR);
                    }
                    return response.setComplete();
                });
    }
}

File: spring-webflux-kotlin/src/main/kotlin/com/example/springwebfluxkotlin/router/ApiRouterConfig.kt
package com.example.springwebfluxkotlin.router

import com.example.springwebfluxkotlin.handler.ApiHandler
import kotlinx.coroutines.handleCoroutineException
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpStatus
import org.springframework.web.reactive.function.client.WebClientResponseException
import org.springframework.web.reactive.function.server.ServerResponse
import org.springframework.web.reactive.function.server.buildAndAwait
import org.springframework.web.reactive.function.server.coRouter

@Configuration
class ApiRouterConfig(private val apiHandler: ApiHandler) {

    @Bean
    fun apiRouter() = coRouter {
        "/api".nest {
            GET("/anime/{id}", apiHandler::getAnimeAndMovie)
            GET("/movie", apiHandler::getMovies)
        }

        onError<Exception> { e, _ ->
            when (e) {
                is WebClientResponseException -> ServerResponse.status(e.rawStatusCode).buildAndAwait()
                else -> ServerResponse.status(HttpStatus.INTERNAL_SERVER_ERROR).buildAndAwait()
            }
        }
    }
}